### PR TITLE
Add User side dish selection functionality

### LINKED ISSUE #173: Get side dishes for a specific mensa and date
To display side dishes for a main dish an API endpoint have to be implemented to return all side dishes for a specific mensa and date

### PR DESCRIPTION
**Pull request by martenjostmann**
closes #173

### CODE REVIEW

File: backend/mensa/views.py [hunk 1]
```diff
@@ -481,3 +478,62 @@ def get_quiz_dishes(request):
...
+        If side or main dish is not in database: 404
+    """
+
+    if 'dishes' in request.data:
+        dishes = request.data['dishes']
+
+        for dish in dishes:
+            if 'main' in dish and 'side' in dish:
+                main = dish['main']
+                side = dish['side']
```
Comment (erikzimmermann) [inline backend/mensa/views.py:521]: What about a list of side dishes? E.g. when you take two side dishes...

**Review by erikzimmermann — CHANGES_REQUESTED** [review]

File: backend/mensa_recommend/serializers.py [hunk 2]
```diff
@@ -85,12 +85,41 @@ class DishPlanSerializer(serializers.ModelSerializer):
...
     class Meta:
         fields = ["dish", "ext_ratings", "user_ratings", "mensa", "date",
-                  "priceStudent", "priceEmployee"]
+                  "priceStudent", "priceEmployee", "side_dishes", "side_selected"]
         model = mensa_model.DishPlan
 
+    def __init__(self, *args, **kwargs):
+        """"
+            Decide which fields should be returned based on the context
+            When the context 'include_sides' is True then side dishes should be included.
```
Comment (erikzimmermann) [inline backend/mensa_recommend/serializers.py:101]: Does side dishes (in general) mean that it just forwards the available side dishes? And selected therefore that the returned sides are only selected ones?

**Review by erikzimmermann — CHANGES_REQUESTED** [review]

File: backend/mensa_recommend/serializers.py (hunk 2 above)
Comment (martenjostmann) [reply backend/mensa_recommend/serializers.py:101]: I don't really understand your question.

File: backend/mensa_recommend/serializers.py [hunk 3]
```diff
@@ -85,12 +85,41 @@ class DishPlanSerializer(serializers.ModelSerializer):
...
         method_name="get_user_ratings")
+    side_dishes = serializers.SerializerMethodField(
+        method_name="get_side_dishes")
+    side_selected = serializers.SerializerMethodField(
+        method_name="get_side_selected")
 
     class Meta:
         fields = ["dish", "ext_ratings", "user_ratings", "mensa", "date",
-                  "priceStudent", "priceEmployee"]
+                  "priceStudent", "priceEmployee", "side_dishes", "side_selected"]
```
Comment (erikzimmermann) [inline backend/mensa_recommend/serializers.py:95]: I'm relatively sure the variable "fields" is a static one... Did you test multiple runs where you first have a context that deletes e.g. the name "side_dishes" and then another run where you want "side_dishes" included?

My guess is that "fields" will be the old instance without "side_dishes".

**Review by erikzimmermann — CHANGES_REQUESTED** [review]

File: backend/mensa_recommend/serializers.py (hunk 3 above)
Comment (martenjostmann) [reply backend/mensa_recommend/serializers.py:95]: The output looks as expected. So I guess this is not a problem. For instance in the first run "side_selected" will be deleted and in the recursive run "side_selected" is not deleted. Same for "side_dishes". I would assume that if "fields" is really static a deletion of an instance results in an error.

**Review by erikzimmermann — APPROVED** [review]

**Pull request merged**